zerver/lib/subscription_info.py

```diff
@@ -521,6 +521,7 @@ def get_subscribers_query(
 def bulk_get_subscriber_peer_info(
     realm: Realm,
     streams: Collection[Stream] | QuerySet[Stream],
+    users_with_metadata_access_via_permission_groups: dict[int, set[int]] | None = None,
```

Comment:
Passing this here so that we do not calculate this twice once here and then again in `send_stream_creation_events_for_previously_inaccessible_streams` when calling `bulk_add_subscriptions`.